cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/SecurityAuthenticationHttpHandler.java

```diff
@@ -201,10 +202,20 @@ private void stopWatchWait(JsonArray externalAuthenticationURIs) throws Exceptio
       port = configuration.getInt(Constants.Security.AUTH_SERVER_BIND_PORT);
     }
 
+    String announceURLs = configuration.get(Constants.Security.AUTH_SERVER_ANNOUNCE_URLS);
+    // If the announceURLs is set in configuration, add the URL's to the list
+    if (announceURLs != null) {
```

Comment:
If this is missing, you should look at the `security.auth.server.announce.address` as well, since it is only deprecated, not removed.